### PR TITLE
Make payout labels unique

### LINKED ISSUE #3421: "Multiple" pull payout transaction check will only show 1 completed transaction
**Describe the bug**
When processing multiple pull payout and checking on the transaction using wallet > transaction detail  (beside label) not all transaction included in that pull payout is showing.

**To Reproduce the bug**
1. Create multiple pull payout and fill the pull payment with bitcoin address
2.  Go to payments payouts > select all the pending pull payment > send and approve multiple "awaiting for approval" payouts
Screen Shot 2022-02-06 at 3 00 43 PM

3. Go to wallet check the transaction using the transaction details icon beside the label
Screen Shot 2022-02-06 at 2 56 55 PM

**Expected behavior**
Only 1 pull payout will show in the completed even if you processed multiple payouts
Screen Shot 2022-02-06 at 2 57 51 PM

**Your BTCPay Environment (please complete the following information):**
- BTCPay Server Version:   1.4.3
- Deployment Method:       docker
- Browser:                  chrome
-

### PR DESCRIPTION
**Pull request by Kukks**
fixes #3421

### CODE REVIEW

**Comment by NicolasDorier** [comment]
I don't think this fix his problem. This PR is just including the ID of the pull payment in the payout label. That will be quite ugly as the label are meant to show short text, not a 70 characters blob.

**Comment by Kukks** [comment]
> I don't think this fix his problem. This PR is just including the ID of the pull payment in the payout label. That will be quite ugly as the label are meant to show short text, not a 70 characters blob.
> 
> If anything we should change the link to the info to show a list of payout rather than only one.

it is a way to fix it as the text needs to be unique to not override each other. But yes agree, it is ugly

**Comment by NicolasDorier** [comment]
My point is that there should be a unique Payout label, but the link to it, when clicked, should send to a page showing the list of payouts.

File: BTCPayServer/HostedServices/TransactionLabelMarkerHostedService.cs [hunk 1]
```diff
@@ -70,37 +72,58 @@ protected override async Task ProcessEvent(object evt, CancellationToken cancell
...
+                        {
+                            await _walletRepository.SetWalletTransactionInfo(updateTransactionLabel.WalletId,
+                                pair.Key.ToString(), walletTransactionInfo);
+                        }
+                    }));
+
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e);
```
Comment (NicolasDorier) [inline BTCPayServer/HostedServices/TransactionLabelMarkerHostedService.cs:124]: remove debug code

File: BTCPayServer/Services/Labels/LabelFactory.cs [hunk 2]
```diff
@@ -33,32 +35,37 @@ WalletRepository walletRepository
...
             }
         }
 
         public IEnumerable<ColoredLabel> GetWalletColoredLabels(WalletBlobInfo walletBlobInfo, HttpRequest request)
         {
-            foreach (var kv in walletBlobInfo.LabelColors)
-            {
-                yield return CreateLabel(new RawLabel() { Text = kv.Key }, kv.Value, request);
-            }
+            return walletBlobInfo.LabelColors.SelectMany(kv => CreateLabel(new RawLabel() { Text = kv.Key }, kv.Value, request));
```
Comment (NicolasDorier) [inline BTCPayServer/Services/Labels/LabelFactory.cs:50]: please don't do this sort of rewrite, it complicates the review and doesn't improve readability

File: BTCPayServer/HostedServices/TransactionLabelMarkerHostedService.cs (hunk 1 above)
Comment (Kukks) [reply BTCPayServer/HostedServices/TransactionLabelMarkerHostedService.cs]: done

**Pull request closed without merging**